File: kotlin/lib/src/main/kotlin/Svix.kt
package com.svix.kotlin

import SvixOptions
import okhttp3.HttpUrl.Companion.toHttpUrlOrNull

class Svix(token: String, options: SvixOptions = SvixOptions()) {

    val application: Application
    val authentication: Authentication
    val endpoint: Endpoint
    val eventType: EventType
    val integration: Integration
    val message: Message
    val messageAttempt: MessageAttempt
    val statistics: Statistics
    val operationalWebhookEndpoint: OperationalWebhookEndpoint

    init {
        val tokenParts = token.split(".")
        if (options.baseUrl == null) {
            val region = tokenParts.last()
            when (region) {
                "us" -> options.baseUrl = "https://api.us.svix.com"
                "eu" -> options.baseUrl = "https://api.eu.svix.com"
                "in" -> options.baseUrl = "https://api.in.svix.com"
                else -> options.baseUrl = "https://api.svix.com"
            }
        }
        val parsedUrl = options.baseUrl?.toHttpUrlOrNull() ?: throw Exception("Invalid base url")
        val defaultHeaders =
            mapOf("User-Agent" to "svix-libs/${Version}/kotlin", "Authorization" to "Bearer $token")
        val httpClient = SvixHttpClient(parsedUrl, defaultHeaders, options.retrySchedule)
        application = Application(httpClient)
        authentication = Authentication(httpClient)
        endpoint = Endpoint(httpClient)
        eventType = EventType(httpClient)
        integration = Integration(httpClient)
        message = Message(httpClient)
        messageAttempt = MessageAttempt(httpClient)
        statistics = Statistics(httpClient)
        operationalWebhookEndpoint = OperationalWebhookEndpoint(httpClient)
    }
}

File: ruby/lib/svix.rb
# frozen_string_literal: true
# This file is @generated

require "json"
require "openssl"
require "base64"
require "uri"
require "logger"

# API
require "svix/api/application"
require "svix/api/authentication"
require "svix/api/background_task"
require "svix/api/endpoint"
require "svix/api/environment"
require "svix/api/event_type"
require "svix/api/health"
require "svix/api/integration"
require "svix/api/message"
require "svix/api/message_attempt"
require "svix/api/operational_webhook_endpoint"
require "svix/api/statistics"

# Models
require "svix/models/aggregate_event_types_out"
require "svix/models/app_portal_access_in"
require "svix/models/app_portal_access_out"
require "svix/models/app_usage_stats_in"
require "svix/models/app_usage_stats_out"
require "svix/models/application_in"
require "svix/models/application_out"
require "svix/models/application_patch"
require "svix/models/application_token_expire_in"
require "svix/models/background_task_out"
require "svix/models/background_task_status"
require "svix/models/background_task_type"
require "svix/models/connector_in"
require "svix/models/connector_kind"
require "svix/models/dashboard_access_out"
require "svix/models/endpoint_headers_in"
require "svix/models/endpoint_headers_out"
require "svix/models/endpoint_headers_patch_in"
require "svix/models/endpoint_in"
require "svix/models/endpoint_message_out"
require "svix/models/endpoint_out"
require "svix/models/endpoint_patch"
require "svix/models/endpoint_secret_out"
require "svix/models/endpoint_secret_rotate_in"
require "svix/models/endpoint_stats"
require "svix/models/endpoint_transformation_in"
require "svix/models/endpoint_transformation_out"
require "svix/models/endpoint_update"
require "svix/models/environment_in"
require "svix/models/environment_out"
require "svix/models/event_example_in"
require "svix/models/event_type_from_open_api"
require "svix/models/event_type_import_open_api_in"
require "svix/models/event_type_import_open_api_out"
require "svix/models/event_type_import_open_api_out_data"
require "svix/models/event_type_in"
require "svix/models/event_type_out"
require "svix/models/event_type_patch"
require "svix/models/event_type_update"
require "svix/models/expung_all_contents_out"
require "svix/models/integration_in"
require "svix/models/integration_key_out"
require "svix/models/integration_out"
require "svix/models/integration_update"
require "svix/models/list_response_application_out"
require "svix/models/list_response_background_task_out"
require "svix/models/list_response_endpoint_message_out"
require "svix/models/list_response_endpoint_out"
require "svix/models/list_response_event_type_out"
require "svix/models/list_response_integration_out"
require "svix/models/list_response_message_attempt_out"
require "svix/models/list_response_message_endpoint_out"
require "svix/models/list_response_message_out"
require "svix/models/list_response_operational_webhook_endpoint_out"
require "svix/models/message_attempt_out"
require "svix/models/message_attempt_trigger_type"
require "svix/models/message_endpoint_out"
require "svix/models/message_in"
require "svix/models/message_out"
require "svix/models/message_status"
require "svix/models/operational_webhook_endpoint_headers_in"
require "svix/models/operational_webhook_endpoint_headers_out"
require "svix/models/operational_webhook_endpoint_in"
require "svix/models/operational_webhook_endpoint_out"
require "svix/models/operational_webhook_endpoint_secret_in"
require "svix/models/operational_webhook_endpoint_secret_out"
require "svix/models/operational_webhook_endpoint_update"
require "svix/models/ordering"
require "svix/models/recover_in"
require "svix/models/recover_out"
require "svix/models/replay_in"
require "svix/models/replay_out"
require "svix/models/status_code_class"
require "svix/models/template_out"

# Core
require "svix/api_error"
require "svix/errors"
require "svix/svix"
require "svix/util"
require "svix/version"
require "svix/webhook"
require "svix/http_error_out"
require "svix/http_validation_error"
require "svix/validation_error"
require "svix/svix_http_client.rb"
require "svix/internal"
